### PR TITLE
ipn: add c2n endpoint for sockstats logs

### PR DESCRIPTION
**Pull request by willnorris**
Add c2n endpoint for collecting sockstats logs from a device.

### CODE REVIEW

**Comment by willnorris** [comment]
This might still be more granular than we were imagining reporting to control, I'm not sure.  This also has the side effect (because of how filch works) that logs are removed from disk when they are read.  So if we were planning to allow local browsing of the on-device logs also, then we need to adjust this approach (likely to not use filch as-is).

**Review by mihaip — COMMENTED** [review]
High-level thoughts, we should probably talk through them at 1pm:
- Since this is an on-demand/debugging endpoint, it seems preferable to do any kind of coalescing/analysis on the control side, where we have more flexibility to change it without having to deploy new clients/get them updated.
- Having this be a lossy process seems not great, as far as a seemingly-read endpoint having side effects. Perhaps one alternate design to consider is that the c2n API does a flush to the logcatcher instead (in a new bucket), and then visualization tooling can read from there. That way we can repeatedly query the data without having to build a bespoke storage system for it. I assume we can configure the bucket with a different (shorter) retention period since it's more sensitive and higher-volume. @dsnet probably has opinions here.

**Comment by willnorris** [comment]
okay, simplified this to just write all current sockstat logs in the c2n response with no event coalescing.  We'll look at adjusting this to use logtail in the future, putting logs into a separate bucket.

File: ipn/ipnlocal/c2n.go [hunk 1]
```diff
@@ -83,6 +83,8 @@ func (b *LocalBackend) handleC2N(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 		writeJSON(res)
+	case "/sockstats":
+		b.sockstatLogger.WriteLogs(w)
```
Comment (mihaip) [inline ipn/ipnlocal/c2n.go:87]: Should probably do a `w.Header().Set("Content-Type", "text/plain")` to ensure the browser doesn't try to interpret the output as HTML.

**Review by mihaip — APPROVED** [review]

File: ipn/ipnlocal/c2n.go (hunk 1 above)
Comment (willnorris) [reply ipn/ipnlocal/c2n.go:88]: done

**Pull request merged**